[Doc] Taichi spec: assignment expressions (#4725)
taichi-dev/taichi: docs/lang/articles/reference.md

### Boolean operations

### Assignment expressions

```

> assignment_expression ::=  [identifier ":="] expression

Reviewer: ```suggestion
assignment_expression ::= [identifier ":="] expression
```

Reviewer: nit: solve conflicts

optional in all other cases. A single expression without a trailing comma
is evaluated to the value of that expression.

> ### Assignment statements

Reviewer: ```suggestion
### Assignment expressions
```
And it should be put between "boolean operations" and "conditional expressions".

Reviewer: The position is still incorrect.

Reviewer: My bad. It seems that making "conditional expressions" immediately after "boolean operations" is more natural. Let's put "assignment expressions" after "conditional expressions".

```
assignment_expression ::=  [identifier ":="] expression
```

An assignment expression assigns an expression to an identifier (see
[assignment statements](#assignment-statements) for more details),
while also returning the value of the expression.

Example:
```python
@ti.kernel
def foo() -> ti.i32:
    b = 2 + (a := 5)
    b += a
    return b
# the return value should be 12
```

:::note
This operator is supported since Python 3.8.
:::

## Simple statements

### Expression statements

> ### Assignment statements

Reviewer: This should not be accidentally removed.

### Assignment statements

> ```

Reviewer: "assignment expression" should be added as an individual subsection under "expressions"

assignment_expression ::=  [identifier ":="] expression
```

> An assignment expression (sometimes also called a “named expression” or “walrus”) assigns an expression to an identifier,

Reviewer: ```suggestion
An assignment expression assigns an expression to an identifier (see
[assignment statements](#assignment-statements) for more details),
```

while also returning the value of the expression.

> 

Reviewer: Add a small example here?

:::note

> This operator is supported in Python 3.8, see [PEP 572](https://www.python.org/dev/peps/pep-0572) for more details about assignment expressions

Reviewer: ```suggestion
This operator is supported since Python 3.8.
```
